### PR TITLE
Refactor Pro Micro pin mapping

### PR DESCRIPTION
**Pull request by Nicell**
This PR refactors the Pro Micro pin mapping:

- Adds in the missing pins 18-21
- Renames `pro_micro_d` to `pro_micro` (`pro_micro_d` alias included for compatibility)
- Removes all uses of `pro_micro_a` in favor of `pro_micro` 
- Updates documentation image + description of Pro Micro pins

### CODE REVIEW

File: docs/docs/development/boards-shields-keymaps.md [hunk 1]
```diff
@@ -47,6 +47,6 @@ in the `app/boards/shields/${board_name}` directory, e.g. `app/boards/shields/cl
...
-  - A [chosen](https://docs.zephyrproject.org/2.5.0/guides/dts/intro.html#aliases-and-chosen-nodes) node named `zmk,kscan` which references the configured KSCAN driver (usually a GPIO matrix). For these keyboards, to be compatible with any Pro Micro compatible boards, the KSCAN configuration should reference the [nexus node](https://docs.zephyrproject.org/2.5.0/guides/porting/shields.html#gpio-nexus-nodes) that ZMK has standardized on. In particular, the `&pro_micro_a` and `&pro_micro_d` aliases can be used to reference the standard `A#` and `D#` pins in shields.
+  - A [chosen](https://docs.zephyrproject.org/2.5.0/guides/dts/intro.html#aliases-and-chosen-nodes) node named `zmk,kscan` which references the configured KSCAN driver (usually a GPIO matrix). For these keyboards, to be compatible with any Pro Micro compatible boards, the KSCAN configuration should reference the [nexus node](https://docs.zephyrproject.org/2.5.0/guides/porting/shields.html#gpio-nexus-nodes) that ZMK has standardized on. In particular, the `&pro_micro` aliases can be used to reference the standard pins of a Pro Micro for shields.
```
Comment (petejohanson) [inline docs/docs/development/boards-shields-keymaps.md:50]:
```suggestion
  - A [chosen](https://docs.zephyrproject.org/2.5.0/guides/dts/intro.html#aliases-and-chosen-nodes) node named `zmk,kscan` which references the configured KSCAN driver (usually a GPIO matrix). For these keyboards, to be compatible with any Pro Micro compatible boards, the KSCAN configuration should reference the [nexus node](https://docs.zephyrproject.org/2.5.0/guides/porting/shields.html#gpio-nexus-nodes) that ZMK has standardized on. In particular, the `&pro_micro` aliases can be used to reference the standard digital pins of a Pro Micro for shields.
```

**Review by petejohanson — CHANGES_REQUESTED** [review]
Minor nitpick. Appreciate the backwards compatible approach, and getting all existing shields updated to help set proper examples moving forward.

File: docs/docs/development/boards-shields-keymaps.md (hunk 1 above)
Comment (Nicell) [reply docs/docs/development/boards-shields-keymaps.md]: Added, thanks!

**Review by petejohanson — COMMENTED** [review]
Thoughts on how to prevent new shields getting merged in w/ the old names? Add an item to the check-list to make sure we're keeping an eye out for it?

**Comment by Nicell** [comment]
> Thoughts on how to prevent new shields getting merged in w/ the old names? Add an item to the check-list to make sure we're keeping an eye out for it?

It doesn't look like there's a way to add a deprecation warning to our nexus nodes in DT, so no way to really detect usage. A check-list item seems like the best idea for now.

**Review by petejohanson — APPROVED** [review]
Thanks! This is much better.

**Pull request merged**